src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGBotManager.cs

```diff
@@ -0,0 +1,237 @@
+    }
+
+    public void OnOverlayClick()
+    {
+        RGDebug.LogVerbose("Showing RG Overlay Menu");
+        selectionPanel.SetActive(true);
+    }
+
+    public void OnOverlayClosed()
+    {
+        selectionPanel.SetActive(false);
+    }
+
+    public void AddBot()
+    {
+        if (_selectedBotPrefab == null)
+        {
+            RGBots.SpawnBot(_selectedBehavior);
+        }
+        else
+        {
+            RGBots.SpawnBot(_selectedBehavior, _selectedBotPrefab);
+        }
+
+        if (_closeOverlayOnBotStart)
+        {
+            OnOverlayClosed();
+        }
+    }
+
+    public void StopAllBots()
+    {
+        var activeBots = GetComponentsInChildren<RGBotEntry>();
+        for (int i = activeBots.Length - 1; i >= 0; i--)
+        {
+            activeBots[i].Delete();
+        }
+    }
+
+    public void SetCloseOnBotStart()
+    {
+        _closeOverlayOnBotStart = !_closeOverlayOnBotStart;
+    }
+
+    public void SetCvRecording()
+    {
+        if (!_cvRecording)
+        {
+            // close this before we switch the flag
+            OnOverlayClosed();
+        }
+        _cvRecording = !_cvRecording;
+    }
+
+    public void AddActiveBot(GameObject runtimeObject, string behavior)
+    {
+        var uiObject = GameObject.Instantiate(rgBotEntry, activeBotRoot.transform);
+        uiObject.GetComponent<RGBotEntry>().Initialize(runtimeObject, behavior);
+    }
+
+    void AddPrefabsToDropdown()
+    {
+        List<string> dropdownOptions = new List<string>();
+        dropdownOptions.Add("Empty");
+        foreach (var prefab in botPrefabs)
+        {
+            dropdownOptions.Add(prefab.name);
+        }
+
+        gameObjectsDropdown.AddOptions(dropdownOptions);
+    }
+
+    void AddBehaviorsToDropdown()
+    {
+        List<string> dropdownOptions = new List<string>();
+
+        var botList = Resources.Load<IRGBotList>("RGBotList");
+        if (!botList)
+        {
+            RGDebug.LogWarning("Failed to load RGBotList from Resources");
+            return;
+        }
+
+        _botEntries = botList.botEntries;
+        foreach (var bot in _botEntries)
+        {
+            dropdownOptions.Add(bot.botName);
+        }
+
+        if (dropdownOptions.Count == 0)
+        {
+            dropdownOptions.Add("Empty");
+        }
+        behaviorsDropdown.AddOptions(dropdownOptions);
+    }
+
+    void PrefabSelected(TMP_Dropdown dropdown)
+    {
+        int index = dropdown.value;
+
+        // Check if "Empty" is selected
+        if (index == 0)
+        {
+            _selectedBotPrefab = null;
+        }
+        else
+        {
+            _selectedBotPrefab = botPrefabs[index - 1];
+        }
+    }
+
+    void BehaviorSelected(TMP_Dropdown dropdown)
+    {
+        int index = dropdown.value;
+
+        // Check if "Empty" is selected
+        _selectedBehavior = null;
+        if (_botEntries.Length > 0)
+        {
+            _selectedBehavior = _botEntries[index - 1].botName;
```

Comment:
Robb, this still doesn't work.  It is selecting the wrong index.

Reply:
Should be fixed in the latest